src/main/java/staffconnect/model/meeting/MeetingDateTime.java

```diff
@@ -6,17 +6,76 @@
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Locale;
 
 /**
  * Represents a Meeting's starting time in the staff book.
  * Guarantees: immutable; is valid as declared in {@link #isValidMeetDateTime(String)}
  */
 public class MeetingDateTime {
 
-    public static final String MESSAGE_CONSTRAINTS = "DateTime should be of the correct format and values dd/mm/yyyy "
-            + "HH:mm";
+    public static final String MESSAGE_CONSTRAINTS = "DateTime should be of the correct format and values\n"
+            + "accepted formats for dates include yyyy-MM-dd, yyyy-M-d, dd-MM-yyyy, yyyy-MM-d, "
+            + "d-MM-yyyy, d-M-yyyy, dd-M-yyyy, d/MM/yyyy, d-M-yyyy, dd-M-yyyy, dd/MM/yyyy, "
+            + "yyyy/MM/dd, yyyy/MM/d, yyyy/M/dd, yyyy/M/d\n"
+            + "accepted formats for time include HH:mm, H:mm, HHmm";
     public static final String VALIDATION_REGEX = "\\d{2}/\\d{2}/\\d{4} \\d{2}:\\d{2}";
 
+    private static final List<String> DATE_FORMATS = List.of(
+            // Separators: -
+            "d-M-uuuu", // d-M-yyyy
+            "dd-M-uuuu", // dd-M-yyyy
+            "d-MM-uuuu", // d-MM-yyyy
+            "dd-MM-uuuu", // dd-MM-yyyy
+            "uuuu-M-d", // yyyy-M-d
+            "uuuu-MM-d", // yyyy-MM-d
+            "uuuu-MM-dd", // yyyy-MM-dd
+
+            // Separators: /
+            "d/M/uuuu", // d/M/yyyy
+            "dd/M/uuuu", // dd/M/yyyy
+            "d/MM/uuuu", // d/MM/yyyy
+            "dd/MM/uuuu", // dd/MM/yyyy
+            "uuuu/M/d", // yyyy/M/d
+            "uuuu/M/dd", // yyyy/M/dd
+            "uuuu/MM/d", // yyyy/MM/d
+            "uuuu/MM/dd" // yyyy/MM/dd
+    );
+
+    private static final List<String> TIME_FORMATS = List.of(
+            "HH:mm",
+            "H:mm",
+            "HHmm"
+    );
+
+    private static final List<String> DATE_REGEXS = List.of(
+            // Separators: -
+            "\\d{1,2}-\\d{1,2}-\\d{4}", // d-M-yyyy
+            "\\d{2}-\\d{1,2}-\\d{4}", // dd-M-yyyy
+            "\\d{1,2}-\\d{2}-\\d{4}", // d-MM-yyyy
+            "\\d{2}-\\d{2}-\\d{4}", // dd-MM-yyyy
+            "\\d{4}-\\d{1,2}-\\d{1,2}", // yyyy-M-d
+            "\\d{4}-\\d{2}-\\d{1,2}", // yyyy-MM-d
+            "\\d{4}-\\d{2}-\\d{2}", // yyyy-MM-dd
+
+            // Separators: /
+            "\\d{1,2}/\\d{1,2}/\\d{4}", // d/M/yyyy
+            "\\d{2}/\\d{1,2}/\\d{4}", // dd/M/yyyy
+            "\\d{1,2}/\\d{2}/\\d{4}", // d/MM/yyyy
+            "\\d{2}/\\d{2}/\\d{4}", // dd/MM/yyyy
+            "\\d{4}/\\d{1,2}/\\d{1,2}", // yyyy/M/d
+            "\\d{4}/\\d{1,2}/\\d{2}", // yyyy/M/dd
+            "\\d{4}/\\d{2}/\\d{1,2}", // yyyy/MM/d
+            "\\d{4}/\\d{2}/\\d{2}" // yyyy/MM/dd
+    );
+
```

Comment:
This may not be the most efficient way to perform the regex validation, as a lot of them has shared similarities and can be compressed down with the | operator , or is redundant. But I don't think this will affect or is in the scope of grading so I think you can let this be.

Reply:
There indeed are redundant expressions. The rationale is to have them correspond to the formats. So the function can iterate only once to verify regex and isParsable simultaneously. Or else the implementation would be more complicated.